api/filter/network/http_connection_manager.proto

```diff
@@ -80,6 +80,22 @@ message HttpConnectionManager {
     // populate the tag name, and the header value is used to populate the tag value. The tag is
     // created if the specified header name is present in the request's headers.
     repeated string request_headers_for_tags = 2;
+
+    // Global target percentage of requests that will be force traced if the
+    // x-client-trace-id header is set. Must be an integer number between 0 and
+    // 100.
+    uint32 client_enabled = 3 [(validate.rules).uint32.lte = 100];
```

Comment:
You will need to use wrapped types, i.e. `google.protobuf.UInt32Value`, to obtain the defaulting behavior specified in https://www.envoyproxy.io/docs/envoy/latest/configuration/http_conn_man/runtime.html?highlight=sample. This is because proto3 can't distinguish between non-specified scalar values and their null values (0 in the case of `uint32`).

Reply:
Ok, so just so we're all clear, you're saying they need to be nullable so that we can detect when they're _not_ set when we plumb it through to the runtime?

Makes sense. Fixed.

Reply:
Yep.